openstack/swift/templates/swift-health-exporter-service.yaml

```diff
@@ -0,0 +1,30 @@
+# This service is needed because the port used in the Prometheus annotation is
+# different from the one used in the proxy-service.
+{{- range $cluster_id, $cluster := .Values.clusters }}
+kind: Service
+apiVersion: v1
+metadata:
+  name: swift-health-exporter-{{$cluster_id}}
+  labels:
+    release: "{{$.Release.Name}}"
+    os-cluster: {{$cluster_id}}
+  annotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "9520"
+    prometheus.io/targets: {{ .Values.alerts.prometheus | quote }}
+spec:
+  selector:
+    component: swift-proxy-{{$cluster_id}}
+  ports:
+    - name: api
+      port: {{$cluster.proxy_public_port}}
+      targetPort: 443
+    {{- if and $cluster.proxy_public_http_port $cluster.sans_http }}
+    - name: api-http
+      port: {{$cluster.proxy_public_http_port}}
+      targetPort: 80
+    {{- end}}
+  externalIPs:
+    - {{$cluster.proxy_public_ip}}
```

Comment:
I am quite unsure here, if you expose the swift API the second time. Did you tried to overwrite the prom annotation in the proxy-service? There is an additional scrape annotation on pod level and so you might not need this additional service.

Reply:
This issue has been resolved over Slack discussion. Tests ongoing in QA.